_search-plugins/knn/knn-index.md

```diff
@@ -135,12 +136,59 @@ An example method definition that specifies an encoder may look something like t
   }
 }
 ```
+**Note** - Method `hnsw` supports `pq` encoder only from OpenSearch 2.10
 
 Encoder name | Requires training | Description
 :--- | :--- | :---
 `flat` | false | Encode vectors as floating point arrays. This encoding does not reduce memory footprint.
 `pq` | true | Short for product quantization, it is a lossy compression technique that encodes a vector into a fixed size of bytes using clustering, with the goal of minimizing the drop in k-NN search accuracy. From a high level, vectors are broken up into `m` subvectors, and then each subvector is represented by a `code_size` code obtained from a code book produced during training. For more details on product quantization, here is a [great blog post](https://medium.com/dotstar/understanding-faiss-part-2-79d90b1e5388)!
 
+These are few other example method definitions of faiss methods using encoders:
+
+* method `ivf` using `flat` encoder
+```json
+"method": {
+  "name":"ivf",
+  "engine":"faiss",
+  "space_type": "l2",
+  "parameters":{
+    "nlist": 4,
+    "nprobes": 2
+  }
+}
+```
+
+* method `ivf` using `pq` encoder
+```json
+"method": {
+  "name":"ivf",
+  "engine":"faiss",
+  "space_type": "l2",
+  "parameters":{
+    "encoder":{
+      "name":"pq",
+      "parameters":{
+        "code_size": 8,
+        "m": 8
+      }
+    }
+  }
+}
+```
+
+* method `hnsw` using `flat` encoder
```

Comment:
No need to include flat encoders I would say

Reply:
I have renamed it. Pls let me know if it requires any modifications.